src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/KVStoRCMigrationActorStateProvider.cs

```diff
@@ -0,0 +1,700 @@
+        async Task IStateProviderReplica.ChangeRoleAsync(ReplicaRole newRole, CancellationToken cancellationToken)
+        {
+            await ((IStateProviderReplica)this.rcStateProvider).ChangeRoleAsync(newRole, cancellationToken);
+            if (newRole == ReplicaRole.Primary)
+            {
+                this.stateProviderInitTask = this.StartStateProviderInitialization(cancellationToken);
+            }
+            else
+            {
+                await this.CancelStateProviderInitializationAsync();
+            }
+        }
+
+        /// <inheritdoc/>
+        public Task CloseAsync(CancellationToken cancellationToken)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).CloseAsync(cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public void Abort()
+        {
+            ((IStateProviderReplica)this.rcStateProvider).Abort();
+        }
+
+        /// <inheritdoc/>
+        public Task BackupAsync(Func<BackupInfo, CancellationToken, Task<bool>> backupCallback)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).BackupAsync(backupCallback);
+        }
+
+        /// <inheritdoc/>
+        public Task BackupAsync(BackupOption option, TimeSpan timeout, CancellationToken cancellationToken, Func<BackupInfo, CancellationToken, Task<bool>> backupCallback)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).BackupAsync(option, timeout, cancellationToken, backupCallback);
+        }
+
+        /// <inheritdoc/>
+        public Task RestoreAsync(string backupFolderPath)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).RestoreAsync(backupFolderPath);
+        }
+
+        /// <inheritdoc/>
+        public Task RestoreAsync(string backupFolderPath, RestorePolicy restorePolicy, CancellationToken cancellationToken)
+        {
+            return ((IStateProviderReplica)this.rcStateProvider).RestoreAsync(backupFolderPath, restorePolicy, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public Task OnSnapshotAsync(TimeSpan currentLogicalTime)
+        {
+            return ((VolatileLogicalTimeManager.ISnapshotHandler)this.rcStateProvider).OnSnapshotAsync(currentLogicalTime);
+        }
+
+        /// <summary>
+        /// Modifies data from KVS store into a suitable format for RC. Saves the modified data in RC store.
+        /// </summary>
+        /// <param name="kvsData">
+        /// Data from KVS store that needs to be modified and saved in RC
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Cancellation token
+        /// </param>
+        /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+        public async Task<long> SaveStateAsync(List<KeyValuePair> kvsData, CancellationToken cancellationToken)
+        {
+            // TODO: Add noise level tracing to emit each kv pair for debugging purposes.
```

Comment:
Even this TODO is resolved. #369 is to emit just keys alone. #434 for instance, emits all the deserailized reminder values. This will help to debug issues in converting KVS data to RC data.